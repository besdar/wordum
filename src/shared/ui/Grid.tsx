import React from 'react';
import {View, StyleSheet, FlexAlignType, FlexStyle} from 'react-native';
import {Surface} from 'react-native-paper';

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    width: '100%',
  },
  growed: {
    flexGrow: 1,
  },
});

type Props = {
  children: React.ReactNode;
  fillAwailableSpace?: boolean;
  alignItems?: FlexAlignType;
  justifyContent?: FlexStyle['justifyContent'];
  direction?: FlexStyle['flexDirection'];
  rowGap?: number;
  gap?: number;
  columnGap?: number;
  padding?: number;
  flexBasis?: FlexStyle['flexBasis'];
  wrap?: boolean;
  surfaceProps?: Omit<React.ComponentProps<typeof Surface>, 'children'>;
};

export const Grid = ({
  children,
  fillAwailableSpace,
  alignItems,
  justifyContent = 'flex-start',
  direction = 'row',
  rowGap,
  columnGap,
  gap,
  padding,
  flexBasis,
  wrap,
  surfaceProps,
}: Props) => {
  const Component = surfaceProps ? Surface : View;

  return (
    <Component
      {...(surfaceProps || {})}
      style={[
        surfaceProps?.style,
        styles.container,
        {
          alignItems,
          justifyContent,
          flexDirection: direction,
          rowGap,
          columnGap,
          gap,
          padding,
          flexBasis,
          flexWrap: wrap ? ('wrap' as const) : ('nowrap' as const),
        },
        fillAwailableSpace && styles.growed,
      ].filter(Boolean)}>
      {children}
    </Component>
  );
};
