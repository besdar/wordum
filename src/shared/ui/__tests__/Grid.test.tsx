import {render} from '@testing-library/react-native';
import React from 'react';
import {Grid} from '../Grid';

describe('FormContainer', () => {
  it('should match the snapshot', () => {
    const {toJSON} = render(<Grid>Test</Grid>);

    expect(toJSON()).toMatchSnapshot();
  });
});
