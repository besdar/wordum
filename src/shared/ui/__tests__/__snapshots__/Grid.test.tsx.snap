// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContainer should match the snapshot 1`] = `
<View
  style={
    [
      {
        "display": "flex",
        "width": "100%",
      },
      {
        "alignItems": undefined,
        "columnGap": undefined,
        "flexBasis": undefined,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "gap": undefined,
        "justifyContent": "flex-start",
        "padding": undefined,
        "rowGap": undefined,
      },
    ]
  }
>
  Test
</View>
`;
